pkg/ipcache/metadata_test.go

```diff
@@ -103,7 +103,7 @@ func TestRemoveLabelsFromIPs(t *testing.T) {
 	assert.NotNil(t, id)
 	assert.Equal(t, 1, id.ReferenceCount)
 	// Simulate adding CIDR policy.
-	ids, err := IPIdentityCache.AllocateCIDRsForIPs([]net.IP{net.ParseIP("1.1.1.1")}, nil)
+	ids, err := IPIdentityCache.AllocateCIDRsForIPs([]net.IP{net.ParseIP("1.1.1.1").To4()}, nil)
```

Comment:
Why do we need this conversion? Is it because we convert this IP to `netip.AddrFromSlice` somewhere? In that case, can't we call `ip.AddrFromIP` internally? Otherwise, caller need to take care of this conversion everywhere they call `AllocateCIDRsForIPs`.

Reply:
Currently all of the callers of this function across the codebase only understand `net.IP`. In order to ship incremental changes and avoid refactoring everything all at once, I drew the line arbitrarily here that I would not change the function signature for `AllocateCIDRsForIPs()`. Definitely over time we can get rid of this ugliness also by updating the callers to use the `netip` types. One step at a time :+1:

Reply:
Probably at some point we can add some tool that greps for the old `net.IP` types and raises a warning / lint error so we can remove all remaining references. For now I was mostly focused on being able to use the new types in https://github.com/cilium/cilium/pull/20116 , so the scope of the changes in this PR are rather limited by that immediate goal.